vavr/src/test/java/com/baeldung/vavr/future/FutureUnitTest.java

```diff
@@ -0,0 +1,212 @@
+    public void givenFunctionReturnInteger_WhenCallWithFuture_ShouldReturnFunctionValue() {
+        Future<Integer> future = Future.of(() -> 1);
+
+        assertEquals(1, future.get()
+            .intValue());
+    }
+
+    @Test
+    public void givenFunctionGetRemoteHttpResourceAsString_WhenCallSuccessWithFuture_ShouldReturnContentValueAsString() {
+        String url = "http://resource";
+        String content = "Content from " + url;
+        Future<String> future = Future.of(() -> getResource(url));
+
+        assertEquals(content, future.get());
+    }
+
+    @Test
+    public void givenFunctionThrowException_WhenCallWithFuture_ShouldReturnFailure() {
+        Future<String> future = Future.of(() -> getResourceThrowException(""));
+        future.await();
+
+        assertTrue(future.isFailure());
+    }
+
+    @Test
+    public void givenAFutureReturnZero_WhenCheckFutureWithExistEvenValue_ShouldReturnRight() {
+        Future<Integer> future = Future.of(() -> 2);
+        boolean result = future.exists(i -> i % 2 == 0);
+
+        assertTrue(result);
+    }
+
+    @Test
+    public void givenFunction_WhenCallWithFutureAndRegisterConsumerForSuccess_ShouldCallConsumerToStoreValue() {
+        final int[] store = new int[] { 0 };
+        Future<Integer> future = Future.of(() -> 1);
+        future.onSuccess(i -> {
+            store[0] = i;
+        });
+        while (store[0] == 0) {
+        }
+
+        assertEquals(1, store[0]);
+    }
+
+    @Test
+    public void givenFunctionThrowException_WhenCallWithFutureAndRegisterConsumerForFailer_ShouldCallConsumerToStoreException() {
+        final Throwable[] store = new Throwable[] { null };
+        Future<String> future = Future.of(() -> getResourceThrowException(""));
+        future.onFailure(err -> store[0] = err);
+        while (store[0] == null) {
+        }
+
+        assertEquals(store[0].getClass(), RuntimeException.class);
+    }
+
+    @Test
+    public void givenAFuture_WhenAddAndThenConsumer_ShouldCallConsumerWithResultOfFutureAction() {
+        int[] store1 = new int[1];
+        int[] store2 = new int[1];
+        Future<Integer> future = Future.of(() -> 1);
+        future.andThen(i -> store1[0] = i.get() + 1)
+            .andThen(i -> store2[0] = store1[0] + 1);
+        while (store2[0] == 0) {
+        }
+
+        assertEquals(2, store1[0]);
+        assertEquals(3, store2[0]);
+    }
+
+    @Test
+    public void givenAFailureFuture_WhenCallOrElseFunction_ShouldReturnNewFuture() {
+        Future<Integer> future = Future.failed(new RuntimeException());
+        Future<Integer> future2 = future.orElse(Future.of(() -> 2));
+
+        assertEquals(2, future2.get()
+            .intValue());
+    }
+
+    @Test
+    public void givenAFuture_WhenCallCancel_ShouldStopTheActionOfFuture() {
+        long waitTime = 1000;
+        int[] store = new int[1];
+        Future<Integer> future = Future.of(() -> {
+            Thread.sleep(waitTime);
+            store[0] = 1;
```

Comment:
It should be possible to obtain the result through Future's API and not by doing this hack